core/runtime/binaryen/runtime_api/core_impl.hpp

```diff
@@ -23,7 +23,8 @@ namespace kagome::runtime::binaryen {
 
     ~CoreImpl() override = default;
 
-    outcome::result<primitives::Version> version() override;
+    outcome::result<primitives::Version> version(
+        std::optional<primitives::BlockHash> const &block_hash) override;
```

Comment:
Please use boost optional. Reason is that old MacOS versions cannot use std::optional

Reply:
We have declared cx17, or not? How does old Mac works with the other constructions?

Reply:
old MacOS also works badly with std::variant and std::filesystem. That's weird thing, although they claim they support C++17, they do not allow to use some parts of its std unless users update their versions of MacOS